crates/notion-core/src/project.rs

```diff
@@ -227,29 +225,37 @@ impl Project {
         path
     }
 
-    /// Writes the specified version of Node to the `toolchain.node` key in package.json.
-    pub fn pin_node_in_toolchain(&self, node_version: NodeVersion) -> Fallible<()> {
-        // update the toolchain node version
-        let toolchain = serial::Image::new(
-            node_version.runtime.to_string(),
-            node_version.npm.to_string(),
-            self.manifest().yarn_str().clone());
-        Manifest::update_toolchain(toolchain, self.package_file())?;
-        println!("Pinned node to version {} in package.json", node_version.runtime);
-        Ok(())
-    }
-
-    /// Writes the specified version of Yarn to the `toolchain.yarn` key in package.json.
-    pub fn pin_yarn_in_toolchain(&self, yarn_version: Version) -> Fallible<()> {
-        // update the toolchain yarn version
-        if let Some(image) = self.manifest().platform() {
-            let toolchain =
-                serial::Image::new(image.node.runtime.to_string(), image.node.npm.to_string(), Some(yarn_version.to_string()));
-            Manifest::update_toolchain(toolchain, self.package_file())?;
-            println!("Pinned yarn to version {} in package.json", yarn_version);
-        } else {
-            throw!(NoPinnedNodeVersion::new());
+    /// Writes the specified version of Node or Yarn to the `toolchain` in package.json.
+    pub fn pin_in_toolchain(&self, distro_version: &DistroVersion) -> Fallible<()> {
```

Comment:
Maybe at this point the method could just be called `pin`?

Reply:
fixed